### DISCUSSION TITLE
how to secure websocket connection

### DISCUSSION THREAD

**Discussion by kingter-sutjiadi** [inspircd/inspircd#2131] [Support] [answered]
hi just wandering how to secure websocket connection so not all user can join my connection only from my APP and WEB only, can i give middleware before go trough the connection?

**Comment by sadiepowell** [comment, marked as answer]
You need to configure a WebSocket origin. See https://docs.inspircd.org/4/modules/websocket/#wsorigin for more information.

**Reply by kingter-sutjiadi** [reply]
hi thank you for the answer, so if i use native ios/android APP how do the APP access the ws ? do i need to set origin also ?

**Reply by sadiepowell** [reply]
Origins are only applicable to web applications. Non-web IRC applications use TCP directly instead of connecting using the WebSocket protocol.

**Reply by kingter-sutjiadi** [reply]
so if the non-web irc application connect using WS protocol when i already set the wsorigin it won't be affected and still able to connect, right?

thank you,

anyway is there any method to secure the TCP connection ?

**Reply by sadiepowell** [reply]
Non-web clients are not subject to origin filtering because its pointless for them as they can just lie about their origin. If you want to secure a server you should require accounts.

**Reply by kingter-sutjiadi** [reply]
hi, can you show me the documentation about "require accounts" 

Thank you 🙏

**Reply by sadiepowell** [reply]
https://docs.inspircd.org/4/modules/account/#connect

You will need to set up a pre-connect class with no account requirement for users to be given time to authenticate something like this:

https://docs.inspircd.org/4/configuration/useful-snippets/#requiring-connections-to-use-sasl

**Reply by kingter-sutjiadi** [reply]
ok thanks a lot @SadieCat 👍 🙏
